covariance/AdaptiveMCMCHandler.cpp

```diff
@@ -0,0 +1,231 @@
+#include "covariance/AdaptiveMCMCHandler.h"
+
+namespace adaptive_mcmc{
+
+// ********************************************
+AdaptiveMCMCHandler::AdaptiveMCMCHandler() {
+// ********************************************
+  start_adaptive_throw  = 0;
+  start_adaptive_update = 0;
+  end_adaptive_update   = 1;
+  adaptive_update_step  = 1000;
+
+  par_means = {};
+  adaptive_covariance = nullptr;
+}
+
+// ********************************************
+AdaptiveMCMCHandler::~AdaptiveMCMCHandler() {
+// ********************************************
+  if(adaptive_covariance != nullptr) {
+    delete adaptive_covariance;
+  }
+}
+
+
+// ********************************************
+void AdaptiveMCMCHandler::InitFromConfig(const YAML::Node& adapt_manager, const std::string& matrix_name_str, const int Npars) {
```

Comment:
Given Npars won't change does it not make sense to make this a class attribute rather than calling it in every method?